frontend/awx/resources/inventories/InventoryForm.tsx

```diff
@@ -86,45 +100,36 @@ export function CreateInventory() {
 
 export function EditInventory() {
   const { t } = useTranslation();
-  const navigate = useNavigate();
+  const pageNavigate = usePageNavigate();
   const params = useParams<{ id?: string }>();
   const id = Number(params.id);
   const { data: inventory } = useGet<Inventory>(`/api/v2/inventories/${id.toString()}/`);
   const { data: igResponse } = useGet<AwxItemsResponse<InstanceGroup>>(
     `/api/v2/inventories/${id.toString()}/instance_groups/`
   );
   // Fetch instance groups associated with the inventory
-  const instanceGroups = igResponse?.results;
+  const originalInstanceGroups = igResponse?.results;
   const onSubmit: PageFormSubmitHandler<InventoryFields> = async (values) => {
-    const { inventory: editedInventory } = values;
-    if (editedInventory.summary_fields.organization.name) {
-      try {
-        const organization = await getOrganizationByName(
-          editedInventory.summary_fields.organization.name
-        );
-        if (!organization) throw new Error(t('Organization not found.'));
-        editedInventory.organization = organization.id;
-      } catch {
-        throw new Error(t('Organization not found.'));
-      }
-    }
+    const { organization, labels, instanceGroups, ...rest } = values;
+
     // Update the inventory
     const updatedInventory = await requestPatch<Inventory>(
       `/api/v2/inventories/${id.toString()}/`,
-      editedInventory
+      { ...rest, organization: organization.id }
     );
 
     // Update inventory with selected instance groups
-    await submitInstanceGroups(updatedInventory, values.instanceGroups ?? [], instanceGroups ?? []);
+    await submitInstanceGroups(
+      updatedInventory,
+      instanceGroups ?? [],
+      originalInstanceGroups ?? []
+    );
 
     // Update inventory with selected labels
-    await submitLabels(updatedInventory, editedInventory.summary_fields.labels.results);
-    navigate(
-      RouteObj.InventoryDetails.replace(':inventory_type', 'inventory').replace(
-        ':id',
-        (updatedInventory.id ?? 0).toString()
-      )
-    );
+    await submitLabels(updatedInventory, inventory?.summary_fields.labels.results || []);
```

Comment:
_Re: the E2E failure on editing the inventory:_ I suspect this has introduced an issue in the PATCH request when we attempt to update labels. The request needs to be patched with summary_fields.labels. But `updatedInventory` does not record changes to summary_fields since it is not part of `InventoryFields`.